### ISSUE TITLE
The epwget application has a bug possibly!

### ISSUE THREAD

**Issue by loveqingping** [eunyoung14/mtcp#67]
In function HandleReadEvent, the code is shown below

```
                            wv->headerset = TRUE;
            wv->recv += (rd - (wv->resp_len - wv->header_len));
            rd = (wv->resp_len - wv->header_len);

                            pbuf += (rd - (wv->resp_len - wv->header_len));
```

Pointer pbuf doesn't move any more, It still points to the HTTP header, not the HTTP paryload!

**Comment by ajamshed** [comment]
Apologies for responding this late. Can you please explain the problem in more detail? "pbuf not moving" does not make sense in this code snippet.

**Comment by lybmath** [comment]
I think this is just a small issue caused by the order.

We found and fixed this problem some time ago. When we run epwget to get a very small file from the server, we found that the downloaded file had only a HTTP header without any file contents although the connection is finished successfully. 

Finally, we found the reason in the code snippet @loveqingping mentioned.
"rd = (wu->resp_len - wv->header_len);" should be put after "pbuf += (rd - (wv-resp_len -wv->header_len));". Otherwise, the expression (rd - (wv->resp_len - wv->header_len)) will always be 0, causing that the pbuf keeps unchanged.
